parser.js

```diff
@@ -82,9 +82,8 @@ function generateAST(code, options) {
       extra.errors = [];
     }
 
-    if (options.ecmaFeatures && typeof options.ecmaFeatures === 'object') {
-      // pass through jsx option
-      extra.ecmaFeatures.jsx = options.ecmaFeatures.jsx;
+    if (typeof options.jsx === 'boolean' && options.jsx) {
```

Comment:
This first check shouldn’t be necessary since the code no longer tries to check an object property.